x-pack/plugins/observability_solution/dataset_quality/common/api_types.ts

```diff
@@ -117,6 +115,7 @@ export type DegradedFieldResponse = rt.TypeOf<typeof getDataStreamDegradedFields
 
 export const dataStreamSettingsRt = rt.partial({
   createdOn: rt.union([rt.null, rt.number]), // rt.null is needed because `createdOn` is not available on Serverless
+  integration: rt.union([rt.null, rt.string]), // Not every dataset belongs to an integration
```

Comment:
Are we actually returning `null`? Why not using something like
```
integration: rt.string
```
? I think that would be enough, in the case of `createdOn` we are explicitly setting the value to null in serverless just to know it's not available